signer/src/signer/rrset.c

```diff
@@ -706,7 +706,9 @@ rrset_sign(hsm_ctx_t* ctx, rrset_type* rrset, time_t signtime)
             continue;
         }
         if (rrset->rrtype != LDNS_RR_TYPE_DNSKEY &&
-	    rrset_sigalgo(rrset, zone->signconf->keys->keys[i].algorithm)) {
+	    rrset_sigalgo(rrset, zone->signconf->keys->keys[i].algorithm) && 
+            (zone->signconf->keys->keys[i].publish == 0 || 
+            (zone->signconf->keys->keys[i].zsk == 0 && zone->signconf->keys->keys[i].ksk == 0))) {
```

Comment:
/* Just checking for the active flag is not enough. This key might be active but
 * if we have a signature of a inactive key we are most likely in a 'smooth rollover
 * between those keys. Unfortunately we can't check since we have no easy way
 * to link the signature back to the key. If this key is set to publish skip signing to
 * make sure we are not disrupting this kind of rollover. The cost is not being able
 * to double sign in some of the (non-default) rollover types.
 */